### PR TITLE
Unified all the id_set current validation to id.py and removed redund…

### PR DESCRIPTION
**Pull request by bakatzir**
…ant code

## Status
- [x] In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/32746

## Description
that includes moving the needed code from https://github.com/demisto/demisto-sdk/pull/888/files to id.py and re organising it
and removing the redundant code it produced.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by ChanochShayner** [comment]
@bakatzir Please do not remove the function `open_id_set`, My functions use it.

**Comment by bakatzir** [comment]
> @bakatzir Please do not remove the function `open_id_set`, My functions use it.

done.

**Comment by ChanochShayner** [comment]
> > @bakatzir Please do not remove the function `open_id_set`, My functions use it.
> 
> done.
Also its use in the validate_manager file
Thanks.

File: demisto_sdk/commands/common/hook_validations/id.py [hunk 1]
```diff
@@ -224,7 +226,37 @@ def _is_mapper_incident_types_found(self, mapper_data):
...
+            bool. if all scripts ids of this playbook are valid.
+        """
+        is_valid = True
+        playbook_data_2nd_level = playbook_data.get(list(playbook_data.keys())[0])
+        implemented_scripts_in_playbook = set(playbook_data_2nd_level.get('implementing_scripts', []))
+        if implemented_scripts_in_playbook:
+            # setting initially to false, if the incident types is in the id_set, it will be valid
+            is_valid = False
+            for script in self.script_set:
+                script_name = list(script.keys())[0]
```
[the remark this answers is not in the document]
Comment (bakatzir) [reply demisto_sdk/commands/common/hook_validations/id.py:244]: this is a wrong var name IMO. will fix.

File: demisto_sdk/commands/validate/validate_manager.py [hunk 2]
```diff
@@ -785,11 +786,16 @@ def validate_no_missing_release_notes(self, modified_files, old_format_files, ad
                                                                                    FileType.TEST_SCRIPT,
                                                                                    FileType.DOC_IMAGE})
         if API_MODULES_PACK in packs_that_should_have_new_rn:
+            if not os.path.isfile(self.id_set_path):
+                IDSetCreator(print_logs=False, output=self.id_set_path).create_id_set()
+            with open(self.id_set_path, 'r') as conf_file:
+                id_set = json.load(conf_file)
```
[the remark this answers is not in the document]
Comment (bakatzir) [reply demisto_sdk/commands/validate/validate_manager.py]: Please re visit my changes, hope i did not miss anything.

File: demisto_sdk/commands/validate/validate_manager.py [hunk 3]
```diff
@@ -1213,26 +1220,6 @@ def get_packs(changed_files):
 
         return packs
 
-    @staticmethod
-    def get_id_set_file(skip_id_set_creation, id_set_path):
```
Comment (roysagi) [inline demisto_sdk/commands/validate/validate_manager.py:1217]: Why are we removing this? we need a new/valid id_set file for the id set checks and dependencies.

**Review by roysagi — COMMENTED** [review]
I wonder if we should change the id set check functionality to run locally as well and not only on circle as well as add a flag to skip it rather than to initiate it, or at least not make it depend on 2 flags (both `--id-set` and `--post-commit`) as it does now.

that way a user that wants to get the interdependencies errors could get them locally and not wait for the build results.

@bakatzir @yaakovi - what do you think?

File: demisto_sdk/commands/validate/validate_manager.py (hunk 3 above)
Comment (bakatzir) [reply demisto_sdk/commands/validate/validate_manager.py:1217]: We are creating it if we pass the correct flag. so this funcis redundant.
```
# id_set validation
        if self.id_set_validations:
            id_set_validator = IDSetValidator(is_circle=self.is_circle, configuration=Configuration(),
                                              ignored_errors=pack_error_ignore_list,
                                              print_as_warnings=self.print_ignored_errors)
            if not id_set_validator.is_file_valid_in_set(file_path, file_type):
                return False
```

**Comment by ChanochShayner** [comment]
@bakatzir you removed all uses of id_set_file from the validate_manager file and I use them in my PR.
Can you please return them?
Thanks.
@roysagi fyi

**Comment by bakatzir** [comment]
@ChanochShayner I'm not meging this until you will merge yours.
as I will fix your PR changes in this one.

**Review by yaakovi — COMMENTED** [review]
looks good to me.

> I wonder if we should change the id set check functionality to run locally as well and not only on circle as well as add a flag to skip it rather than to initiate it, or at least not make it depend on 2 flags (both --id-set and --post-commit) as it does now.
> 
> that way a user that wants to get the interdependencies errors could get them locally and not wait for the build results.
agree, whatever we test in the build should also be easily tested locally.

**Comment by ChanochShayner** [comment]
@roysagi @bakatzir 
It should be remembered that there are cases where one of the entities is in the private repo and then we need to verify the validation with the unified ID set, and it is not possible to do this in a local run.
In case we run the local validate it will return an error, because it will not find the other entity, and the real thing is that there is no error here but only running in with the unified id set will know it and can only be done in build.

**Comment by guykeller** [comment]
@bakatzir May I close this, or you would like to keep working on it when you get the time?

**Comment by bakatzir** [comment]
@guykeller I plan to continue after aviya will merge her PR.

**Comment by bakatzir** [comment]
Won't get to it. closing.

**Pull request closed without merging**